src/builtin/pre_commit_hooks/check_added_large_files.rs

```diff
@@ -0,0 +1,73 @@
+use crate::git::intent_to_add_files;
+use crate::hook::Hook;
+use clap::Parser;
+use futures::StreamExt;
+use std::collections::{HashMap, HashSet};
+
+enum FileFilter {
+    NoFilter,
+    Files(HashSet<String>),
+}
+
+impl FileFilter {
+    fn contains(&self, path: &str) -> bool {
+        match self {
+            FileFilter::NoFilter => true,
+            FileFilter::Files(files) => files.contains(path),
+        }
+    }
+}
+
+#[derive(Parser)]
+struct Args {
+    #[arg(long)]
+    #[arg(help = "Enforce all files are checked, not just staged files.")]
+    enforce_all: bool,
+    #[arg(default_value = "500")]
+    #[arg(help = "Maximum allowable KB for added files")]
+    max_kb: u64,
+}
+
+pub(crate) async fn check_added_large_files(
+    hook: &Hook,
+    filenames: &[&String],
+    _env_vars: &HashMap<&'static str, String>,
+) -> anyhow::Result<(i32, Vec<u8>)> {
+    let entry = shlex::split(&hook.entry).ok_or(anyhow::anyhow!("Failed to parse entry"))?;
+    let args = Args::try_parse_from(entry.iter().chain(&hook.args))?;
+
+    let filter = if args.enforce_all {
+        FileFilter::NoFilter
+    } else {
+        let add_files: HashSet<_> = intent_to_add_files().await?.into_iter().collect();
+        FileFilter::Files(add_files)
+    };
+
+    let mut tasks = futures::stream::iter(filenames.iter().filter(|f| filter.contains(f))).map(
+        async |filename| {
+            let len = fs_err::tokio::metadata(filename).await?.len();
+            if len > args.max_kb {
```

Comment:
I think `len` should be divided by `1024` before it's compared with `args.max_kb`?

Reply:
yes, you are right. I will fix it later